integration/shareddata/scalars.go

```diff
@@ -24,36 +24,37 @@ import (
 )
 
 const (
-	doubleMaxPrec = float64(1<<53 - 1) // 9007199254740991.0:    largest double values that could be represented as integer exactly
-	doubleBig     = float64(1 << 61)   // 2305843009213693952.0: some number larger than safe integer (doubleBig+1 == doubleBig)
-	longBig       = int64(1 << 61)     // 2305843009213693952:   same as doubleBig but integer
+	doubleMaxPrec float64 = float64(1<<53 - 1) // 9007199254740991.0:    largest double values that could be represented as integer exactly
+	doubleBig     float64 = float64(1 << 61)   // 2305843009213693952.0: some number larger than safe integer (doubleBig+1 == doubleBig)
+	longBig       int64   = int64(1 << 61)     // 2305843009213693952:   same as doubleBig but integer
 
-	// TODO https://github.com/FerretDB/FerretDB/issues/2321
+	doubleMaxOverflow float64 = float64(math.MaxInt64) + 1100.0 // 9223372036854777856.0: double value that is larger than MaxInt64
+	doubleMinOverflow float64 = float64(math.MinInt64) - 1100.0 // -9223372036854777856.0: double value that is smaller than MinInt64
 )
 
+var doubleMaxVerge float64 = math.Nextafter(math.MaxFloat64, 0) // 1.7976931348623155e+308: double value that is close to the MaxFloat64 (1.7976931348623157e+308)
+
 // Scalars contain scalar values for tests.
 //
 // This shared data set is frozen. If you need more values, add them in the test itself.
 var Scalars = &Values[string]{
 	name: "Scalars",
 	data: map[string]any{
-		"double":                    42.13,
-		"double-whole":              42.0,
-		"double-zero":               0.0,
-		"double-max":                math.MaxFloat64,
-		"double-smallest":           math.SmallestNonzeroFloat64,
-		"double-big":                doubleBig,
-		"double-1":                  float64(math.MinInt64 - 1),
-		"double-2":                  float64(math.MinInt64),
-		"double-3":                  float64(-123456789),
-		"double-4":                  float64(123456789),
-		"double-5":                  float64(math.MaxInt64),
-		"double-6":                  float64(math.MaxInt64 + 1),
-		"double-7":                  1.79769e+307,
-		"double-max-overflow":       9.223372036854776833e+18, // TODO https://github.com/FerretDB/FerretDB/issues/2321
-		"double-max-overflow-verge": 9.223372036854776832e+18,
-		"double-min-overflow":       -9.223372036854776833e+18,
-		"double-min-overflow-verge": -9.223372036854776832e+18,
+		"double":              42.13,
+		"double-whole":        42.0,
+		"double-zero":         0.0,
+		"double-max":          math.MaxFloat64,
+		"double-smallest":     math.SmallestNonzeroFloat64,
+		"double-big":          doubleBig,
+		"double-1":            float64(math.MinInt64 - 1),
```

Comment:
But we already have a constant for "double value that is smaller than MinInt64" (`doubleMinOverflow`). Why not use it?